pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionMetaDataManager.java

```diff
@@ -111,6 +114,26 @@ public void initialize() {
         }
     }
 
+    static FunctionMetaData normalizeFunctionMetaData(FunctionMetaData fmd) {
+        if (!fmd.getFunctionDetails().getSource().getTopicsToSerDeClassNameMap().isEmpty()) {
+            FunctionDetails.Builder fdb = FunctionDetails.newBuilder(fmd.getFunctionDetails());
+            for (Map.Entry<String, String> topicEntry : fmd.getFunctionDetails().getSource().getTopicsToSerDeClassNameMap().entrySet()) {
+                fdb.getSourceBuilder().putInputSpecs(
+                    topicEntry.getKey(),
+                    ConsumerSpec.newBuilder()
+                        .setSerdeClassName(topicEntry.getValue())
+                        .setIsRegexPattern(topicEntry.getKey() == fmd.getFunctionDetails().getSource().getTopicsPattern())
```

Comment:
this should be false

Reply:
why? if topic patterns was used, this should be true, no?